### PR TITLE
Split build component by service

### PR DESCRIPTION
**Pull request by Michkov**
Separate build component to:
- build-service
- jvm-build-service
- pipeline-service
- shared-resources
- build-templates

### CODE REVIEW

File: components/build/README.md [hunk 1]
```diff
@@ -1,113 +0,0 @@
-
-# App Studio/HACBS Build System
```
Comment (psturc) [inline components/build/README.md:2]: imo this readme file contains useful information - is none of it going to be needed in a future?

Comment (Michkov) [reply components/build/README.md:2]: hmm, probably I could split it as well

Comment (Michkov) [reply components/build/README.md]: just moved to build-service for now.

**Review by bnallapeta — APPROVED** [review]
LGTM. 
A minor comment:
Should be renamed from build-service to pipeline-service [here](https://github.com/Michkov/infra-deployments/blob/build-split/components/pipeline-service/tekton-results/.tekton/pipelines-as-code.yaml#L4-L5)?

**Comment by Michkov** [comment]
> LGTM. A minor comment: Should be renamed from build-service to pipeline-service [here](https://github.com/Michkov/infra-deployments/blob/build-split/components/pipeline-service/tekton-results/.tekton/pipelines-as-code.yaml#L4-L5)?

Updated, moved under tekton-pipelines since pipeline-service namespace does not exist. But Anyway I am thinking to move these .tekton folders under one place and into one namespace in next PRs

**Comment by adambkaplan** [comment]
LGTM from me.

**Comment by tkdchen** [comment]
:+1:

**Review by psturc — APPROVED** [review]

**Pull request merged**